vgen/src/main/java/io/gapi/vgen/py/PythonDiscoveryContext.java

```diff
@@ -0,0 +1,249 @@
+import io.gapi.vgen.DiscoveryImporter;
+
+import java.util.List;
+
+/**
+ * A DiscoveryContext specialized for Python.
+ */
+public class PythonDiscoveryContext extends DiscoveryContext {
+
+  /**
+   * A map from primitive field types used by DiscoveryImporter to Python counterparts.
+   */
+  private static final ImmutableMap<Field.Kind, String> FIELD_TYPE_MAP =
+      ImmutableMap.<Field.Kind, String>builder()
+          .put(Field.Kind.TYPE_UNKNOWN, "dict")
+          // TODO(tcoffee): check validity on actual cases
+          .put(Field.Kind.TYPE_BOOL, "bool")
+          .put(Field.Kind.TYPE_INT32, "int")
+          .put(Field.Kind.TYPE_UINT32, "int")
+          .put(Field.Kind.TYPE_INT64, "long")
+          .put(Field.Kind.TYPE_UINT64, "long")
+          .put(Field.Kind.TYPE_FLOAT, "float")
+          .put(Field.Kind.TYPE_DOUBLE, "float")
+          .put(Field.Kind.TYPE_STRING, "str")
+          .build();
+
+  /**
+   * A map from Python native type name to corresponding default value.
+   */
+  private static final ImmutableMap<String, String> NATIVE_DEFAULT_MAP =
+      ImmutableMap.<String, String>builder()
+          .put("dict", "{}")
+          .put("bool", "False")
+          .put("int", "0")
+          .put("long", "0L")
+          .put("float", "0.0")
+          .build();
+
+  /**
+   * A map from {@link ApiaryConfig#stringFormat} label (or null) to corresponding default value.
+   */
+  private static final ImmutableMap<String, String> STRING_DEFAULT_MAP =
+      ImmutableMap.<String, String>builder()
+          .put("byte", "''"
+              + "  # base64-encoded string of bytes: see http://tools.ietf.org/html/rfc4648")
+          .put("date", "'1969-12-31'"
+              + "  # 'YYYY-MM-DD': see datetime.date.isoformat()")
+          .put("date-time", "'" + new DateTime(0L).toStringRfc3339() + "'"
+              + "  # 'YYYY-MM-DDThh:mm:ss.fffZ' (UTC): see datetime.datetime.isoformat()")
+          .build();
+
+  private static final ImmutableMap<String, String> RENAMED_METHOD_MAP =
+      ImmutableMap.<String, String>builder()
+          .build();
+
+  private final PythonContextCommon pythonCommon;
+
+  /**
+   * Constructs the Python discovery context.
+   */
+  public PythonDiscoveryContext(Service service, ApiaryConfig apiaryConfig) {
+    super(service, apiaryConfig);
+    pythonCommon = new PythonContextCommon();
+  }
+
+  public PythonContextCommon python() {
+    return pythonCommon;
+  }
+
+  // Snippet Helpers
+  // ===============
+
+  public String silent(String any) {
+    return "";
+  }
+
+  public String getSampleVarName(String typeName) {
+    return pythonCommon.wrapIfKeywordOrBuiltIn(upperCamelToLowerUnderscore(typeName));
+  }
+
+  @Override
+  public String getMethodName(Method method) {
+    return getSimpleName(getRename(method.getName(), RENAMED_METHOD_MAP));
+  }
+
+  /**
+   * Returns a name for the type of a field of a type, substituting the given name when a native
```

Comment:
nit: "for a type's field's type"

My own personal preference is here and in a bunch of the comments below to use a string of possessives to go from the outside in, rather than a string of "of"s to go from the inside out.